kornia/geometry/homography.py

```diff
@@ -35,24 +36,22 @@ def find_homography_dlt(points1: torch.Tensor, points2: torch.Tensor, weights: t
     # DIAPO 11: https://www.uio.no/studier/emner/matnat/its/nedlagte-emner/UNIK4690/v16/forelesninger/lecture_4_3-estimating-homographies-from-feature-correspondences.pdf  # noqa: E501
     ax = torch.cat([zeros, zeros, zeros, -x1, -y1, -ones, y2 * x1, y2 * y1, y2], dim=-1)
     ay = torch.cat([x1, y1, ones, zeros, zeros, zeros, -x2 * x1, -x2 * y1, -x2], dim=-1)
+    A = torch.cat((ax, ay), dim=-1).reshape(ax.shape[0], -1, ax.shape[-1])
+
+    if weights is None:
+        # All points are equally important
+        A = A.transpose(-2, -1) @ A
+    else:
+        # We should use provided weights
+        assert len(weights.shape) == 2 and weights.shape == points1.shape[:2], weights.shape
+        w_diag = torch.diag_embed(weights.repeat(1, 2))
+        A = A.transpose(-2, -1) @ w_diag @ A
 
-    w_list = []
-    axy_list = []
-    for i in range(points1.shape[1]):
-        axy_list.append(ax[:, i])
-        axy_list.append(ay[:, i])
-        w_list.append(weights[:, i])
-        w_list.append(weights[:, i])
-    A = torch.stack(axy_list, dim=1)
-    w = torch.stack(w_list, dim=1)
-
-    # apply weights
-    w_diag = torch.diag_embed(w)
-    A = A.transpose(-2, -1) @ w_diag @ A
     try:
         U, S, V = torch.svd(A)
     except:
-        return torch.empty(points1_norm.size(0), 3, 3)
+        print('SVD did not converged')
```

Comment:
`RuntimeWarning('SVD did not converged')
`

Reply:
Git it!